src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1230,9 +1509,15 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
             var local = node.LocalSymbol;
             int slot = GetOrCreateSlot(local);
 
+            if (!node.ArgumentsOpt.IsDefault)
+            {
+                VisitAll(node.ArgumentsOpt);
+            }
+
             var initializer = node.InitializerOpt;
             if (initializer is null)
             {
+                VisitTypeExpression(node.DeclaredType);
```

Comment:
>VisitTypeExpression [](start = 16, length = 19)

Why visit `BoundTypeExpression` instances? Do we report any warnings or set a value in the map other than `node.Type`?

Reply:
We need to get this information for the public API, and some of these `BoundTypeExpression`s will be `var`. So we need to set what we infer for them.

---
In reply to: [265702159](https://github.com/dotnet/roslyn/pull/34017#discussion_r265702159) [](ancestors = 265702159)